src/OrchardCore.Modules/OrchardCore.Media/Drivers/MediaFieldDriver.cs

```diff
@@ -112,6 +124,7 @@ public override async Task<IDisplayResult> UpdateAsync(MediaField field, IUpdate
                 {
                     try
                     {
+                        field.SetAttachedFileNames(items.Where(i => !i.IsRemoved).Select(i => i.AttachedFileName).ToArray());
```

Comment:
Add property to media field and use here

Reply:
@ns8482e  This only applies to attached media. Do you still want the field on [src/OrchardCore/OrchardCore.Media.Core/Fields/MediaField.cs](https://github.com/OrchardCMS/OrchardCore/blob/c02fedd8c3bf1bb1b892e3c5fc0c000d27a9ca80/src/OrchardCore/OrchardCore.Media.Core/Fields/MediaField.cs)?

Reply:
Thank you, I just wanted to make sure.

Reply:
> Add property to media field and use here

I'd suggest not doing this @ns8482e .

The value is only going to be supplied when using the attached media field editor. Not the standard editor.

So the design choice there is to attach things to the field, as extra meta data with extension methods.

Reply:
@DrewBrasher @deanmarcussen sounds good

Reply:
Also using a metadata and not adding a property to the field make it possible to create this as a separate feature. The feature would add the UI to edit and store the metadata. Right now it seems like it's not a separate feature, but at least it's not breaking anything, as long as the value is not mandatory.